py/server/tests/test_vectorization.py

```diff
@@ -0,0 +1,204 @@
+#
+# Copyright (c) 2016-2022 Deephaven Data Labs and Patent Pending
+#
+import random
+import time
+import unittest
+from types import SimpleNamespace
+from typing import List, Any
+
+import deephaven
+
+from deephaven import DHError, read_csv, empty_table, SortDirection, AsOfMatchRule, time_table, ugp
+from deephaven.agg import sum_, weighted_avg, avg, pct, group, count_, first, last, max_, median, min_, std, abs_sum, \
+    var, formula, partition
+from deephaven.execution_context import make_user_exec_ctx
+from deephaven.filters import Filter, and_
+from deephaven.html import to_html
+from deephaven.pandas import to_pandas
+from deephaven.table import Table, _dh_vectorize
+from tests.testbase import BaseTestCase
+
+
+class VectorizationTestCase(BaseTestCase):
+    def setUp(self):
+        super().setUp()
+        deephaven.table._test_vectorization = True
+        deephaven.table._vectorized_count = 0
+
+    def tearDown(self) -> None:
+        deephaven.table._test_vectorization = False
+        deephaven.table._vectorized_count = 0
+        super().tearDown()
+
+    def test_vectorization_exceptions(self):
+        t = empty_table(1)
+
+        @_dh_vectorize
+        def vectorized_func(p1, p2):
+            return p1 + p2
+
+        def auto_func(p1, p2):
+            return p1 + p2
+
+        @_dh_vectorize
```

Comment:
It was not a high conviction decision to make the decorator private. There are pros and cons between marking it as private and marking as public. The benefit of letting the users use it so that an exception would be thrown if the function can't be used in a formula might overweigh the potential problem mentioned in the doc string.